presto-parser/src/main/java/com/facebook/presto/sql/tree/QueryOrderBy.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Collections.emptyList;
+import static java.util.Objects.requireNonNull;
+
+public class QueryOrderBy
+        extends Node
+{
+    public static final QueryOrderBy NO_ORDER_BY = new QueryOrderBy(emptyList());
+
+    private final List<SortItem> sortItems;
+
+    public QueryOrderBy(List<SortItem> sortItems)
```

Comment:
If we make `QueryOrderBy` optional in `QuerySpecification` then we probably can enforce sortItems list to be non-empty.